### ISSUE TITLE
Bell character in .bashrc causes startup hang

### ISSUE THREAD

**Issue by dwierenga** [warpdotdev/Warp#2513]
### Describe the bug

Having a bell character (0x07) in my .bashrc file was enough to cause a hang on startup.
I had this simple line in my rc file:
`alias beep='echo ^G'`
where ^G was an actual 0x7 character. 

### To Reproduce

1.  Add this line to your `.bashrc` file:
`alias beep='echo ^G'`
2. Open a new terminal tab, and watch the startup hang.

### Expected behaviour

The terminal start-up should not hang when encountering a bell character.

### Operating System

MacOS

### OS Version

12.6.2

### Shell Version

bash 5.2.15

### Warp Version

v0.2023.01.10.08.02.stable_01

### Does this block you from using Warp daily?

No

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by elviskahoro** [comment]
@dwierenga i tried this but wasn't able to make Warp hang. What else is in your bash rc? I think it could be a combination of things or something else!

**Comment by dwierenga** [comment]
@elviskahorol I've done a bit more digging.  I commented out everything in `/etc/profile`, and `/etc/bashrc` so that only my `~/.bash_login` and `~/.bashrc` are involved.
In `~/.bash_login` I have this single line:
`[[ -r ~/.bashrc ]] && . ~/.bashrc`

In `~/.bashrc` I have only this line:
`alias beep='echo ^G'`

Commenting out either line in either file allows normal startup.  The combination of the two creates the hang. ¯\\_(ツ)_/¯

**Comment by dwierenga** [comment]
No, of course I'm an idiot.  Commenting out the line in `~/.bash_login` appears to work only because my `~/.bashrc` is then never sourced. Adding the alias directly to `~/.bash_login` causes the hang.  It does appear to be directly isolated to the bell character.

Maybe you don't have a real bell character in your `~/.bashrc` file?  If you run the `beep` alias, does it actually ring the terminal bell?

**Comment by dwierenga** [comment]
This is still happening with v0.2026.04.08.08.36.stable_05.

Reproduced by having this single line in ~/.bash_login:
```
alias beep='echo ^G'
```
where ^G is an actual control character i.e. running `beep` should actually ring your terminal bell.

With this and only this line in my Bash startup files, Warp hangs.
